src/zoo/owl_zoo_cmd.mli

```diff
@@ -6,7 +6,7 @@
 
 (** {6 Manipulate gists} *)
 
-val remove_gist : string -> unit
+val remove_gist : ?all:bool -> string -> unit
```

Comment:
I think the signature should be: `?vid -> string -> unit`, if vid is None, then remove the whole gist, otherwise remove one specific version, right?

What is your string then? Is it just a gist id, or is it the whole name scheme ... such as `gist | vid | ...`

Reply:
Definitely correct. Adding `-a` is only a temporary solution. I think the signature could be kept unchanged, and `remove_gist gist` parses the whole gist name (`gid/vid/pin`) to see if there is a `vid` contained.

Reply:
I think your coding still tangles everything together. Do you really think `remove` a zoo gist and `install/download` a zoo gist should understand the gist string in the same way?

For installing a zoo gist, you can expand the gist id as we discussed before. But for removing, I am not sure what you are doing is correct. What does `pin` mean to `remove`???

Reply:
OK I see, `remove` does not need to understand `pin`, only `gid` and optionally `vid`. My previous idea only makes this function pointlessly complex.

Reply:
The point when you are designing the naming scheme, you need to look in every command need to have the same understanding or not. You can often categorise them into several groups. `remove` imo is not in the same one as `download`. 

I have been asking you to maintain a specification file, explaining the grammar and semantics of the naming and how each key command understands it. I guess you really need it, otherwise your coding is just patching rather than designing.

Reply:
Understand. Put the specification on the todo list. Thanks for helping to fix.